mldsa/fips202/fips202x4.c

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) The mlkem-native project authors
+ * Copyright (c) The mldsa-native project authors
+ * SPDX-License-Identifier: Apache-2.0 OR ISC OR MIT
+ */
+
+#include "../common.h"
+
+#include <string.h>
+#include "fips202.h"
+#include "fips202x4.h"
+#include "keccakf1600.h"
+
+typedef mld_shake128x4ctx mld_shake256x4_ctx;
+
+/*TODO: Proof*/
+static void mld_keccak_absorb_once_x4(uint64_t *s, uint32_t r,
```

Comment:
Seeing you're introducing contracts and invariants already, is adding the proofs not merely a matter of adding harness and Makefile? Maybe we should do it in one go?

Reply:
I've added the proofs now. Please take a look.